admin/server/organizations.go

```diff
@@ -240,79 +249,69 @@ func (s *Server) UpdateOrganizationBillingSubscription(ctx context.Context, req
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	if org.BillingCustomerID == "" {
-		// create new customer
-		customerID, err := s.admin.Biller.CreateCustomer(ctx, org)
-		if err != nil {
-			return nil, status.Error(codes.Internal, err.Error())
+	org, subs, err := s.admin.RepairOrgBilling(ctx, org)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	planChange := true
+	for _, sub := range subs {
+		if sub.Plan.ID == plan.ID {
+			planChange = false
+			break
 		}
-		org.BillingCustomerID = customerID
+	}
 
-		// create new subscription
-		_, err = s.admin.Biller.CreateSubscription(ctx, org.BillingCustomerID, plan)
+	if planChange {
+		c, err := s.admin.Payment.FindCustomer(ctx, org.PaymentCustomerID)
 		if err != nil {
 			return nil, status.Error(codes.Internal, err.Error())
 		}
-	} else {
-		subs, err := s.admin.Biller.GetSubscriptionsForCustomer(ctx, org.BillingCustomerID)
-		if err != nil {
-			return nil, status.Error(codes.Internal, err.Error())
+		if !c.ValidPaymentMethod {
+			return nil, status.Errorf(codes.FailedPrecondition, "no valid payment method found for the organization")
 		}
 
-		if len(subs) > 0 {
-			for _, sub := range subs {
-				if sub.Plan.ID == plan.ID {
-					return nil, status.Error(codes.InvalidArgument, "same plan already assigned to the organization")
-				}
+		if len(subs) == 1 {
+			// schedule plan change
+			_, err = s.admin.Biller.ChangeSubscriptionPlan(ctx, subs[0].ID, plan)
+			if err != nil {
+				return nil, status.Error(codes.Internal, err.Error())
 			}
-
-			if len(subs) == 1 {
-				// schedule plan change
-				_, err = s.admin.Biller.ChangeSubscriptionPlan(ctx, subs[0].ID, plan)
-				if err != nil {
-					return nil, status.Error(codes.Internal, err.Error())
-				}
-			} else {
-				// multiple subscriptions, cancel them first immediately and assign new plan
-				// should not happen unless externally assigned multiple subscriptions to the same org in the billing system
-				for _, sub := range subs {
-					err = s.admin.Biller.CancelSubscription(ctx, sub.ID, billing.SubscriptionCancellationOptionImmediate)
-					if err != nil {
-						return nil, status.Error(codes.Internal, err.Error())
-					}
-				}
-
-				// create new subscription
-				_, err = s.admin.Biller.CreateSubscription(ctx, org.BillingCustomerID, plan)
+		} else {
+			// multiple subscriptions, cancel them first immediately and assign new plan
+			// should not happen unless externally assigned multiple subscriptions to the same org in the billing system
+			for _, sub := range subs {
```

Comment:
Is this needed now that it calls `RepairOrgBilling` above?

Reply:
So repair org does not fix multiple subscription issue, as I am not sure which subscription to cancel and which to keep, I can add a warn or error log in this case and hook it up so that we get notified properly. However this API is plan change API so it can just cancel all older subscription and create a new one with new plan.

Reply:
Got it. Might be worth adding a comment about.